mailbox/api/src/main/java/org/apache/james/mailbox/MessageManager.java

```diff
@@ -172,6 +175,10 @@ public Long getSize() {
             return messageAttachments.get();
         }
 
+        public ThreadId getThreadId() {
+            return threadId;
+        }
+
         @Override
         public final boolean equals(Object o) {
             if (o instanceof AppendResult) {
```

Comment:
We do not keep threadId into account for equals & hashCode?

Is that POJO equalsVerifer tested?

Can you fix it?